src/types/redis_stream.cc

```diff
@@ -232,6 +241,35 @@ std::string Stream::encodeStreamConsumerMetadataValue(const StreamConsumerMetada
   return dst;
 }
 
+StreamConsumerMetadata Stream::decodeStreamConsumerMetadataValue(const std::string &value) {
+  StreamConsumerMetadata consumer_metadata;
+  rocksdb::Slice input(value);
+  GetFixed64(&input, &consumer_metadata.pending_number);
+  GetFixed64(&input, &consumer_metadata.last_idle);
+  GetFixed64(&input, &consumer_metadata.last_active);
+  return consumer_metadata;
+}
+
+StreamSubkeyType Stream::identifySubkeyType(const rocksdb::Slice &key) {
+  InternalKey ikey(key, storage_->IsSlotIdEncoded());
+  Slice subkey = ikey.GetSubKey();
+  const size_t entry_id_size = sizeof(StreamEntryID);
+  if (subkey.size() <= entry_id_size) {
+    return StreamSubkeyType::StreamEntry;
+  }
+  uint64_t group_name_len = 0;
+  GetFixed64(&subkey, &group_name_len);
+  std::string without_group_name = subkey.ToString().substr(group_name_len);
+  const size_t metadata_delimiter_size = strlen(consumerGroupMetadataDelimiter);
```

Comment:
You can make `consumerGroupMetadataDelimiter` a `string_view` and then remove this `strlen`.